smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/config/DefaultClientConfiguration.kt

```diff
@@ -55,6 +55,7 @@ class DefaultClientConfiguration : ClientConfiguration {
             parameters = listOf(
                 FunctionParameter.NoLabel("provider", ClientRuntimeTypes.Core.InterceptorProvider)
             ),
+            isMutating = true,
```

Comment:
Places the `mutating` modifier on these methods, since they now change properties of a value type